File: Graduating-Server/src/main/kotlin/com/bestswlkh0310/graduating/graduatingserver/core/meal/MealType.kt
package com.bestswlkh0310.graduating.graduatingserver.core.meal

enum class MealType {
    BREAKFAST,
    LUNCH,
    DINNER;

    companion object {
        fun ofKorean(string: String) = entries.firstOrNull { it.korean() == string }
    }

    fun korean() = when (this) {
        BREAKFAST -> "조식"
        LUNCH -> "중식"
        DINNER -> "석식"
    }
}

File: Graduating-iOS/Shared/Model/Meal/MealType.swift
//
//  MealType.swift
//  Model
//
//  Created by hhhello0507 on 8/28/24.
//

import Foundation

public enum MealType: String, ModelProtocol {
    case breakfast = "BREAKFAST"
    case lunch = "LUNCH"
    case dinner = "DINNER"
    
    public var korean: String {
        switch self {
        case .breakfast: "조식"
        case .lunch: "중식"
        case .dinner: "석식"
        }
    }
    
    
    public static func from(_ date: Date) -> Self? {
        switch (date[.hour], date[.minute]) {
        case (0...8, _), (8, ..<20):
                .breakfast // 아침: ~ 8:20
        case (8, 21...60), (8...13, _), (13, 0..<30):
                .lunch // 점심: 8:21 ~ 13:30
        case (13, 0...30), (13...19, _), (19, 0..<10):
                .dinner // 저녁: 13:31 ~ 19:10
        default: nil
        }
    }
}
